### PR TITLE
Adds more strict linting

### PR DESCRIPTION
**Pull request by AngellusMortis**
Updates existing workflow to also lint the following (in addition to hassfest):

* pylint
* black
* isort
* pyupgrade
* mypy

Updates the code everywhere for the linters as well.

### CODE REVIEW

File: custom_components/unifiprotect/__init__.py [hunk 1]
```diff
@@ -53,10 +51,11 @@
 from .models import UnifiProtectEntryData
 from .services import (
     add_doorbell_text,
+    profile_ws,
     remove_doorbell_text,
     set_default_doorbell_text,
-    profile_ws,
 )
+from .utils import above_ha_version
```
Comment (bdraco) [inline custom_components/unifiprotect/__init__.py:58]: 👍

File: custom_components/unifiprotect/entity.py [hunk 2]
```diff
@@ -45,26 +46,33 @@ def __init__(
             sw_version=self.device.firmware_version,
             connections={(dr.CONNECTION_NETWORK_MAC, self.device.mac)},
             configuration_url=self.device.protect_url,
+            default_manufacturer=DEFAULT_BRAND,
+            default_model="",
+            entry_type=None,
+            identifiers=set(),
+            suggested_area=None,
+            default_name="",
```
Comment (bdraco) [inline custom_components/unifiprotect/entity.py:54]: These should have defaults and shouldn't be needed

Comment (AngellusMortis) [reply custom_components/unifiprotect/entity.py:54]: mypy was complaining about them.

Comment (AngellusMortis) [reply custom_components/unifiprotect/entity.py:54]:
```
custom_components/unifiprotect/entity.py:41: error: Missing keys ("default_manufacturer", "default_model", "default_name", "entry_type", "identifiers", "suggested_area") for TypedDict "DeviceInfo"  [typeddict-item]
```

**Review by bdraco — APPROVED** [review]

**Pull request merged**